pkg/ctl/cmdutils/configfile.go

```diff
@@ -180,12 +180,12 @@ func NewCreateClusterLoader(provider *api.ProviderConfig, spec *api.ClusterConfi
 		}
 
 		return ngFilter.CheckEachNodeGroup(l.spec.NodeGroups, func(i int, ng *api.NodeGroup) error {
-			if ng.AllowSSH && ng.SSHPublicKeyPath == "" {
+			if ng.SSH.Allow && ng.SSH.PublicKeyPath == nil {
```

Comment:
In the case of flags it this should remain as an actual empty string check, actually we might move it down below...

Reply:
What I think we should do is this:
- make `cfg.NewNodeGroup()` should construct `&NodeGroupSSH{Allow: NewBoolFlase(), PublicKeyPath: DefaultSSHPublicKeyPath}`
- `ng.SSH.Allow` gets passed as default value of `--ssh-access`
- `ng.SSH.PublicKeyPath` gets passed as default value of `--ssh-public-key`

The idea is that all of defaulting logic is defined in the API package, albeit it there are `NewClusterConfig` and `NewNodeGroup` for flags and `SetNodeGroupDefaults` for config files, those are still in the same package.

Right now we have this:
https://github.com/weaveworks/eksctl/blob/653cd2a174120ddb38f9025f928543fbe04b3f75/pkg/ctl/cmdutils/nodegroup_flags.go#L39-L40

And based on the above it would become:
```Go
fs.BoolVar(&ng.SSH.Allow, "ssh-access", ng.SSH.Allow, "control SSH access for nodes")
fs.StringVar(&ng.SSH.PublicKeyPath, "ssh-public-key", ng.SSH.PublicKeyPath, "SSH public key to use for nodes (import from local path, or use existing EC2 key pair)")
```

The `nodegroup_flags.go` doesn't implement this pattern everywhere yet, but that's what I've been trying to do for every flag that pairs with a config file filed.